File: Day4/MergeIntervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
 var merge = function (itr) {
    const n = itr.length
    itr.sort((x, y) => x[0] - y[0])
    const result = [[itr[0][0], itr[0][1]]];
    for (let i = 1; i < n; i++) {
        const last = result.length - 1
        if (result[last][1] < itr[i][0]) {
            result.push(itr[i])
        } else {
            result[last][1] = result[last][1] > itr[i][1] ? result[last][1] : itr[i][1]
        }
    }


    return result
};

console.log(merge([[1,4],[4,5]]));

File: Day98/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        n = len(intervals)
        def srt(key):
            return key[0]
        intervals.sort(key=srt)
        result = [intervals[0]]
        for i in range(n):
            last = len(result) - 1
            if result[last][1] < intervals[i][0]:
                result.append(intervals[i])
            else:
                result[last][1] = result[last][1] if result[last][1] > intervals[i][1] else intervals[i][1]
        return result
